core/src/main/scala/org/apache/spark/sql/execution/columnar/JDBCSourceAsStore.scala

```diff
@@ -184,20 +185,21 @@ final class ColumnBatchIteratorOnRS(conn: Connection,
     fetchColQuery: String)
     extends ResultSetIterator[Array[Byte]](conn, stmt, rs, context) {
   var currentUUID: String = _
-  val ps = conn.prepareStatement(fetchColQuery)
-  var colBuffers: Option[scala.collection.mutable.HashMap[Int, (ByteBuffer, Blob)]] = None
+  val ps: PreparedStatement = conn.prepareStatement(fetchColQuery)
+  var colBuffers: Option[TIntObjectHashMap] = None
+
   def getColumnLob(bufferPosition: Int): ByteBuffer = {
     colBuffers match {
-      case Some(map) => map(bufferPosition)._1
+      case Some(map) =>
+        map.get(bufferPosition).asInstanceOf[(ByteBuffer, Blob)]._1
```

Comment:
TIntObjectHashMap is making the code complex and needs some asInstanceOf calls. Given that, the number of entries in the Hash map is going to be not more than the number of columns, would the performance difference be too much?